vars/publishToNpm.groovy

```diff
@@ -7,16 +7,34 @@
  * compatible open source license.
  */
 
-/** Library to publish artifacts to NPM registry under @opensearch-project namespace
+/** Library to publish artifacts to NPM registry under @opensearch-project namespace.
 @param Map args = [:] args A map of the following parameters
-@param args.repository <required> - Repository to be used to publish the artifact to npm
-@param args.tag <required> - Tag reference to be used to publish the artifact
+@param args.publicationType <required> - github (will clone the repository at triggered tag and url and publish), artifact (Needs artifactPath arg to the URL or local artifact that needs to be published)
+@param args.artifactPath <required with publicationType: artifact > - URL or local Path to the artifact that needs to be publish to NPM.See supported artifacts https://docs.npmjs.com/cli/v9/commands/npm-publish?v=true#description for more details.
 */
 void call(Map args = [:]) {
+    allowedPublicationType = ['github', 'artifact']
+    artifactPath = args.artifactPath ?: ''
 
-    checkout([$class: 'GitSCM', branches: [[name: "${args.tag}" ]], userRemoteConfigs: [[url: "${args.repository}" ]]])
-    
-    withCredentials([string(credentialsId: 'jenkins-opensearch-publish-to-npm-token', variable: 'NPM_TOKEN')]){
-        sh """npm set registry "https://registry.npmjs.org"; npm set //registry.npmjs.org/:_authToken ${NPM_TOKEN}; npm publish --dry-run && npm publish --access public"""
+    if (!allowedPublicationType.contains(args.publicationType)) {
+        currentBuild.result = 'ABORTED'
+        error('Invalid publicationType. publicationType can either be github or artifact')
     }
+    if (args.publicationType == 'artifact' && !args.artifactPath) {
+        currentBuild.result = 'ABORTED'
+        error('publicationType: artifact needs an artifactPath. Please provide artifactPath argument. See supported artifacts https://docs.npmjs.com/cli/v9/commands/npm-publish?v=true#description for more details')
+    }
+    if (args.publicationType == 'github') {
```

Comment:
You probably also want to check if type is github, then artifactpath needs to be empty or the npm publish will go back to publishing with tar.

Reply:
Thanks for bringing this up. Added the check. Created a new method just for parameter check to make it cleaner.